components/eam/bld/configure

```diff
@@ -1812,12 +1818,18 @@ if (defined $opts{'use_MMF'}) {
     my $crm_nx_rad = $cfg_ref->get('crm_nx_rad');
     my $crm_ny_rad = $cfg_ref->get('crm_ny_rad');
     my $crm = $cfg_ref->get('crm');
+    my $pam_dycor = $cfg_ref->get('pam_dycor');
     my $MMF_microphysics_scheme = $cfg_ref->get('MMF_microphysics_scheme');
     my $crm_adv = $cfg_ref->get('crm_adv');
     my $yes3Dval = 1;
     if ($crm_ny eq 1) {$yes3Dval = 0;}
     if (defined $opts{'crm_adv'}) { $cfg_cppdefs .= " -D_$crm_adv "; }
-    $cfg_cppdefs .= " -D$MMF_microphysics_scheme -DYES3DVAL=$yes3Dval";
+    $cfg_cppdefs .= " -DYES3DVAL=$yes3Dval";
+    if ($MMF_microphysics_scheme eq 'sam1mom') {
```

Comment:
this maybe a style difference so you may leave it as it is. I would combine if & elsif statement into one since $cfg_cppdefs .= "-D$MMF_microphysics_scheme" is the same.